java/arcs/android/type/ParcelableSchemaFields.kt

```diff
@@ -61,10 +61,22 @@ data class ParcelableSchemaFields(val actual: SchemaFields) : Parcelable {
         else -> throw IllegalStateException("List of unexpected type encountered")
       }
 
+    private fun Parcel.readNullableFieldType(): FieldType =
```

Comment:
Could this function just be:

```
return FieldType.NullableOf(readFieldType())
```

If you wanted to check that the inner type is valid:

```
val innerValue = readFieldType()
require(innerValue.tag == FieldType.Tag.Primitive || innerValue.tag == FieldType.Tag.EntityRef || ...)
return FieldType.NullableOf(innerValue)
```

Reply:
Yep, in retrospect it doesn't really make sense for the encoding + decoding logic to be doing validity checking (that should be in the constructors/builders if anywhere). That makes the code much simpler and easier to follow.

Thanks